packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -136,7 +162,12 @@ - (instancetype)initWithCameraName:(NSString *)cameraName
   _photoIOQueue = dispatch_queue_create("io.flutter.camera.photoIOQueue", NULL);
   _videoCaptureSession = videoCaptureSession;
   _audioCaptureSession = audioCaptureSession;
-  _captureDevice = [AVCaptureDevice deviceWithUniqueID:cameraName];
+  if (captureDevice) {
+    _captureDevice = captureDevice;
+  } else {
+    _captureDevice = [AVCaptureDevice deviceWithUniqueID:cameraName];
```

Comment:
This would move up to the convenience inits (at which point it looks like the `cameraName` parameter can be removed from this inner init method).

Reply:
Since captureDevice is also set in the `setDescriptionWhileRecording` method, I went with another block for creating captureDevice. Got rid of `cameraName`